### PR TITLE
fix: add event emission for ppo transfer hook contracts

### PR DESCRIPTION
**Pull request by Nilay27**
* Adding missing events for `BlocklistTransferHook` and `RestrictedTransferHook` with tests and natspec in interface.

### CODE REVIEW

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides nit!

**Comment by Nilay27** [comment]
* Removed `setToken()` , `getToken()` and `onlyToken()` modifier along with relevant tests.

**Comment by ramenforbreakfast** [comment]
Approved! Just want to clarify these recent changes were added since initially, we were adding `getToken` and `setToken` to our interfaces since they were missing. Now that we've decided to remove them, I made the call to just remove them in this PR to reduce the amount of PRs currently in-flight and since this is already a small PR.

File: apps/smart-contracts/token/contracts/ppo/interfaces/IBlocklistTransferHook.sol [hunk 1]
```diff
@@ -9,6 +9,10 @@ import "./IAccountList.sol";
  * A blocked address cannot send or receive the specified ERC20 token.
  */
 interface IBlocklistTransferHook is ITransferHook {
+  /// @dev Emitted via `setBlocklist()`.
+  /// @param newBlocklist Address of the `IAccountList` contract
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/ppo/interfaces/IBlocklistTransferHook.sol:13]:
```suggestion
  /**
   * @dev Emitted via `setBlocklist()`.
   * @param newBlocklist Address of the `IAccountList` contract
   */
```

File: apps/smart-contracts/token/contracts/ppo/interfaces/IRestrictedTransferHook.sol [hunk 2]
```diff
@@ -16,6 +16,14 @@ import "./IAccountList.sol";
  * Blocklisted addresses cannot send or receive tokens, even if allowlisted.
  */
 interface IRestrictedTransferHook is IBlocklistTransferHook {
+  /// @dev Emitted via `setSourceAllowlist()`.
+  /// @param newSourceAllowlist Address of the `IAccountList` contract
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/ppo/interfaces/IRestrictedTransferHook.sol:20]:
```suggestion
  /** 
   * @dev Emitted via `setSourceAllowlist()`.
   * @param newSourceAllowlist Address of the `IAccountList` contract
   */
```

File: apps/smart-contracts/token/contracts/ppo/interfaces/IRestrictedTransferHook.sol [hunk 3]
```diff
@@ -16,6 +16,14 @@ import "./IAccountList.sol";
...
+  event SourceAllowlistChange(IAccountList newSourceAllowlist);
+
+  /// @dev Emitted via `setDestinationAllowlist()`.
+  /// @param newDestinationAllowlist Address of the `IAccountList` contract
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/ppo/interfaces/IRestrictedTransferHook.sol:24]:
```suggestion
  /**
   * @dev Emitted via `setDestinationAllowlist()`.
   * @param newDestinationAllowlist Address of the `IAccountList` contract
   */
```

**Review by ramenforbreakfast — COMMENTED** [review]
Realized these comments needed to be multiline.

**Review by XavierEkkel — APPROVED** [review]
LGTM

**Pull request merged**